### PR TITLE
Add kubelet_disk_type in AKS additional_node_pools

### PR DESCRIPTION
**Pull request by NissesSenap**
This allows us to use Temporary disk types which should remove most of
the risk of running ful on images, emptyDir and logs on AKS nodes.
Solves #554

Upgrade azurerm to 2.97 to get this new feature.

The valid options for `kubelet_disk_type` are: 
OS and Temporary

### CODE REVIEW

**Comment by NissesSenap** [comment]
If you don't want to change your existing node pool all you need to do is to configure it to use: `kubelet_disk_type = "OS"`

**Comment by NissesSenap** [comment]
We have read lots and lots of docs and we don't think that the kubelet_disk_type is in preview and thus ready for production.

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-01-01/managedClusters.json

For more information:
https://github.com/Azure/AKS/issues/930
https://docs.microsoft.com/en-us/rest/api/aks/agent-pools/create-or-update#kubeletdisktype

**Comment by NissesSenap** [comment]
When trying to apply the config I get the error saying that it's a preview feature. So I think we can skip this for now since it will only force us to set another config value.

```
│ Error: creating Node Pool: (Agent Pool Name "standard3" / Managed Cluster Name "aks-dev-we-aks1" / Resource Group "rg-dev-we-aks"): containerservice.AgentPoolsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="PreviewFeatureNotRegistered" Message="Preview feature Microsoft.ContainerService/KubeletDisk not registered."
│
```

**Comment by NissesSenap** [comment]
As noted in the issue this feature have now gone GA.
An interesting thing to know is that even though terraform say it can do a in-place change it seems like it isn't possible.

I ran a in-place change but after running it a second time I get the following information. 

```
      ~ kubelet_disk_type      = "OS" -> "Temporary"
```

So my guess is that Azure changed back...
I'm currently trying to verify if this is the case when I create a new node pool or not.

**Comment by NissesSenap** [comment]
I have misunderstood, I thought that custom node config ment kubelet disk type since it's a configuration that is done on the node... It's not.

Even though I can't find anything about kubelet disk type https://docs.microsoft.com/en-us/rest/api/aks/managed-clusters/create-or-update#kubeletdisktype being in preview still.
I might miss something obvious in the terraform provider but I'm unsure...

File: modules/azure/aks/variables.tf [hunk 1]
```diff
@@ -36,15 +36,16 @@ variable "aks_config" {
...
-      min_count      = number
-      max_count      = number
-      spot_enabled   = bool
-      spot_max_price = number
-      node_taints    = list(string)
-      node_labels    = map(string)
+      name              = string
+      version           = string
+      vm_size           = string
+      kubelet_disk_type = string
```
Comment (NissesSenap) [inline modules/azure/aks/variables.tf:42]: Should we have this as a temporary variable to be able to migrate easier between versions?
Or should we just remove it as a variable?

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 - [#664](https://github.com/XenitAB/terraform-modules/pull/664) Manage Helm chart CRDs outside of Helm.
 - [#678](https://github.com/XenitAB/terraform-modules/pull/678) Update OPA to 3.8.1, gatekeeper-library to 0.12.1 and add k8srequireingressclass constraint.
+- [#561](https://github.com/XenitAB/terraform-modules/pull/561) [Breaking] Add kubelet_disk_type as a config to AKS additional_node_pools. this allows us to use `Temporary` disk types.
```
Comment (NissesSenap) [inline CHANGELOG.md:33]: Need to update changelog

**Review by NissesSenap — COMMENTED** [review]
It's still a preview feature :( 

```shell
│ Error: creating Node Pool: (Agent Pool Name "standard1" / Managed Cluster Name "aks-qa-we-aks1" / Resource Group "rg-qa-we-aks"): containerservice.AgentPoolsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="PreviewFeatureNotRegistered" Message="Preview feature Microsoft.ContainerService/KubeletDisk not registered."
```

**Comment by landerss1** [comment]
Replaced by [#1152](https://github.com/XenitAB/terraform-modules/pull/1153).

**Pull request closed without merging**